lib/src/barback/dartdevc/dartdevc_environment.dart

```diff
@@ -253,24 +255,26 @@ AssetId _entrypointDartId(AssetId id) {
 class _AssetCache {
   /// [Asset]s are indexed first by package and then path, this allows us to
```

Comment:
I haven't, but I don't think this introduces undue complexity. The extra logic is confined to `_getResult` and `[]=`, and it simplifies `invalidatePackage` a bit.

The number of assets in this map is O(#modules), which isn't generally going to be that large, but if we introduce other module schemes (like one module per strongly connected component) it could blow up fairly quickly.